dt-assets/js/service-worker.js

```diff
@@ -0,0 +1,139 @@
+self.addEventListener("install", () => {
+  self.skipWaiting();
+});
+
+self.addEventListener('notificationclick', (event) => {
+  const linkURL = event.notification.data.link;
+
+  event.notification.close();
+
+  event.waitUntil(
+    clients.matchAll({ type: 'window' }).then((clientList) => {
+      for (const client of clientList) {
+        if (client.url === '/' && 'focus' in client) {
+          return client.focus();
+        }
+      }
+      if (clients.openWindow && linkURL) {
+        return clients.openWindow(linkURL);
+      }
+    })
+  );
+});
+
+addEventListener("message", (event) => {
+
+  const url = '/wp-json/dt/v1/notifications/get_notifications';
```

Comment:
Wordpress sites can be hosted at different root URLs, so we need to make use of `window.wpApiShare.root` as the root of the API request. Sometimes the site is in a subdirectory when hosting a multi-site instance. That root property will be an absolute path to the wp-json directory, like `https://mydt.local/wp-json/`, so you can just append the rest of the notification endpoint to that.

Reply:
@jlamanskygitt This one still needs to be addressed. And it then needs to be applied to the added API request to mark the notification read.